collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -698,6 +705,58 @@ private void nonblockingCreateSavePointData() {
         }
     }
 
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        if (currentView != null) {
+            for (QuestionWidget qw : ((ODKView) currentView).getWidgets()) {
+                if (qw instanceof DateWidget) {
+                    LocalDateTime date = new LocalDateTime()
+                            .withYear(year)
+                            .withMonthOfYear(month + 1)
+                            .withDayOfMonth(dayOfMonth)
+                            .withHourOfDay(0)
+                            .withMinuteOfHour(0)
+                            .withSecondOfMinute(0)
+                            .withMillisOfSecond(0);
+                    ((DateWidget) qw).setBinaryData(date);
+                    widgetValueChanged(qw);
+                    return;
+                } else if (qw instanceof DateTimeWidget) {
+                    LocalDateTime date = new LocalDateTime()
+                            .withYear(year)
+                            .withMonthOfYear(month + 1)
+                            .withDayOfMonth(dayOfMonth)
+                            .withHourOfDay(0)
+                            .withMinuteOfHour(0)
+                            .withSecondOfMinute(0)
+                            .withMillisOfSecond(0);
+                    ((DateTimeWidget) qw).setBinaryData(date);
+                    widgetValueChanged(qw);
+                    return;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
```

Comment:
Also the fact that here you use `onTimeSet` and above `setBinaryData` is not consistent i think it would be good to use `setBinaryData` in all cases. I know that name doesn't match very well since originally it was created for media widgets so I would rename it to just `setAnswer` maybe.

Reply:
I totally agree with you that using `setBinaryData()` at all places will be consistent. I didn't know that we can have two Time widgets on one screen for ODK X Forms. I usually use ODK Build to build testing forms. I didn't see any feature like that to include two Time widgets on one screen there.

Reply:
Theoretically it is possible that's why we use waitForData()/cancelWaitingForData